src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -82,4 +98,21 @@ public void forEach(Consumer<LogFile> callback) {
     public boolean isEmpty() {
         return this.logFiles.isEmpty();
     }
+
+    /**
+     * Get the LogFile object from the fileHash.
+     * @param fileHash the fileHash obtained from uploader.
+     * @return the logFile.
+     */
+    public LogFile getFile(String fileHash) {
+        if (!fileHashToFileMap.containsKey(fileHash)) {
+            logger.atDebug().log("FileHash does not exist");
+        }
+        return fileHashToFileMap.get(fileHash);
+    }
+
+    public LogFileGroup update() throws InvalidLogGroupException {
```

Comment:
Update is not representative of what it is doing. Call is something like snapshot. Also add some doc to the method that explains what it does and maybe why it is useful